mon-pix/tests/helpers/service-stubs.js

```diff
@@ -0,0 +1,103 @@
+import Service from '@ember/service';
+import sinon from 'sinon';
+
+/**
+ * Stub the session service.
+ *
+ * @param {Object} owner - The owner object.
+ * @param {Object} [session={}] - The session object.
+ * @param {boolean} [session.isAuthenticated=false] - Indicates if the user is authenticated.
+ * @param {boolean} [session.isAuthenticatedByGar=false] - Indicates if the user is authenticated by GAR.
+ * @param {string} [session.externalUserTokenFromGar='external-user-token'] - The external user token from GAR.
+ * @param {string} [session.userIdForAssociation='expected-user-id'] - The expected user ID from GAR.
+ * @param {string} [session.source='pix'] - The source of authentication.
+ * @returns {Service} The stubbed session service.
+ */
+export function stubSessionService(owner, session = {}) {
+  const isAuthenticated = session.isAuthenticated || false;
+  const isAuthenticatedByGar = session.isAuthenticatedByGar || false;
+  const externalUserTokenFromGar = session.externalUserTokenFromGar || 'external-user-token';
+  const userIdForAssociation = session.userIdForAssociation;
+  const source = isAuthenticated ? session.source || 'pix' : null;
+
+  /**
+   * Stub class for the session service.
+   */
+  class SessionStub extends Service {
+    constructor() {
+      super();
+      this.isAuthenticated = isAuthenticated;
+      this.isAuthenticatedByGar = isAuthenticatedByGar;
+      this.userIdForAssociation = userIdForAssociation;
+
+      if (isAuthenticated) {
+        this.data = { authenticated: { source } };
+      } else {
+        this.data = {};
+      }
+
+      if (isAuthenticatedByGar) {
+        this.externalUserTokenFromGar = externalUserTokenFromGar;
+      }
+
+      this.authenticate = sinon.stub();
+      this.authenticateUser = sinon.stub();
+      this.requireAuthentication = sinon.stub();
+      this.invalidate = sinon.stub();
+      this.revokeGarExternalUserToken = sinon.stub();
+      this.revokeGarAuthenticationContext = sinon.stub();
+      this.requireAuthenticationAndApprovedTermsOfService = sinon.stub();
+      this.setAttemptedTransition = sinon.stub();
+    }
+  }
+
+  owner.register('service:session', SessionStub);
+  return owner.lookup('service:session');
+}
+
+/**
+ * Stub the current user service.
+ *
+ * @param {Object} owner - The owner object.
+ * @param {Object} [user={}] - The user object.
+ * @param {boolean} [user.isAnonymous=false] - Indicates if the user is anonymous.
+ * @param {string} [user.firstName='John'] - The first name of the user.
+ * @param {string} [user.lastName='Doe'] - The last name of the user.
+ * @param {boolean} [user.mustValidateTermsOfService=false] - Indicates if the user must validate terms of service.
+ * @param {boolean} [user.hasRecommendedTrainings=false] - Indicates if the user has recommended trainings.
+ * @returns {Service} The stubbed current user service.
+ */
+export function stubCurrentUserService(owner, user = {}) {
```

Comment:
Je n'en suis pas sûr mais je propose, je trouve que cela pourrait être plus clair en disant qu'on passe « des données de l'utilisateur », plutôt qu'un « utilisateur », sinon en tant que développeur je vais avoir l'impression qu'on me demande de passer « le vrai utilisateur ».

```suggestion
export function stubCurrentUserService(owner, userData = {}) {
```